exonum-java-binding-core/rust/integration_tests/src/mock/transaction.rs

```diff
@@ -70,44 +72,46 @@ pub fn create_throwing_exec_exception_mock_transaction_proxy(
                 )?
                 .l()?;
             let java_tx_mock = env.new_global_ref(java_tx_mock)?;
-            let raw = RawMessage::new(MessageBuffer::from_vec(vec![]));
+            let raw = create_empty_raw_transaction();
             Ok((java_tx_mock, raw))
         })
         .unwrap();
 
-    TransactionProxy::from_global_ref(executor, java_tx_mock, raw)
+    let tx_proxy = TransactionProxy::from_global_ref(executor, java_tx_mock, raw.clone());
+    (tx_proxy, raw)
 }
 
 /// Creates `TransactionProxy` with a mock transaction and an empty `RawMessage`.
-pub fn create_mock_transaction_proxy(executor: MainExecutor, valid: bool) -> TransactionProxy {
-    let (java_tx_mock, raw) = create_mock_transaction(&executor, valid);
-    TransactionProxy::from_global_ref(executor, java_tx_mock, raw)
+pub fn create_mock_transaction_proxy(
+    executor: MainExecutor,
+    valid: bool,
+) -> (TransactionProxy, RawTransaction) {
+    let (java_tx_mock, raw) = create_mock_transaction(&executor);
+    let tx_proxy = TransactionProxy::from_global_ref(executor, java_tx_mock, raw.clone());
+    (tx_proxy, raw)
 }
 
 /// Creates a mock transaction and an empty `RawMessage`.
-pub fn create_mock_transaction(executor: &MainExecutor, valid: bool) -> (GlobalRef, RawMessage) {
+pub fn create_mock_transaction(executor: &MainExecutor) -> (GlobalRef, RawTransaction) {
     executor
         .with_attached(|env| {
             let value = env.new_string(ENTRY_VALUE)?;
-            let info = env.new_string(INFO_JSON)?;
             let java_tx_mock = env
                 .call_static_method(
                     NATIVE_FACADE_CLASS,
                     "createTransaction",
-                    format!(
-                        "(ZLjava/lang/String;Ljava/lang/String;)L{};",
-                        TRANSACTION_ADAPTER_CLASS
-                    ),
-                    &[
-                        JValue::from(valid),
-                        JValue::from(JObject::from(value)),
-                        JValue::from(JObject::from(info)),
-                    ],
+                    format!("(Ljava/lang/String;)L{};", TRANSACTION_ADAPTER_CLASS),
+                    &[JValue::from(JObject::from(value))],
                 )?
                 .l()?;
             let java_tx_mock = env.new_global_ref(java_tx_mock)?;
-            let raw = RawMessage::new(MessageBuffer::from_vec(vec![]));
+            let raw = create_empty_raw_transaction();
             Ok((java_tx_mock, raw))
         })
         .unwrap()
 }
+
+pub fn create_empty_raw_transaction() -> RawTransaction {
+    let service_transaction = ServiceTransaction::from_raw_unchecked(0, vec![]);
```

Comment:
I'd consider some locals for clarity (what are zeros).

Reply:
Done